### ISSUE TITLE
Big repo over 60 GB 

### ISSUE THREAD

**Issue by GoogleCodeExporter** [waterxi/svnplot#97]
```
What steps will reproduce the problem?

I have a really big repository about 65 GB.
First commit for this repository is from 2004!
That means it takes very very long until I get the db-file and afterwards the 
graphics.


What is the expected output? What do you see instead?
Is there a possibility to get the statistics from a specific period?! For 
example from 2011 till now or only the last year?!


What version of the product are you using? 

python-2.7.9rc1
py27-pysvn-svn185-1.7.9-1572
numpy-1.9.1-win32-superpack-python2.7
matplotlib-1.4.2.win32-py2.7
SVNPlot-0.8.4.win32 => to get the data from svn repo
SVNPlot-0.7.10.win32 => to generate the graphs, with SVNPlot-0.8.4 I get an 
error "UnicodeEncodeError: 'ascii' codec can't encode characters"

On what operating system?
Windows 7 x64

Thx in advance.


```

Original issue reported on code.google.com by `Har00n...@gmail.com` on 22 Jan 2015 at 7:12

**Comment by GoogleCodeExporter** [comment]
```
Because of the way data is stored in svn, to get reliable statistics requires 
scanning and importing entire data once in the sqlite database. Once the data 
is sqlite database, svnplotjs.py or svnplot.py has start revision/end revision 
parameters to extract data from specific period.

There are two ways to reduce the initial import time
1. Copy the repository somewhere so that it is accessible as 'file:///' url. 
And then run the svnlog2sqlite.py. This way, 'delta' is not transfered over 
network and hence it faster.
2. Use 'svndump' and 'svndumpfilter' command to create a partial copy of 
repository with the paths/dates that you need. Then run svnlog2sqlite on this 
new copy.  However this may also take long time.

Also please send me the call stack for 'UnicodeEncodeError'. You will find call 
stack in the log file. So that I will fix it.
```

Original comment by `nitinbh...@gmail.com` on 22 Jan 2015 at 1:36

**Comment by GoogleCodeExporter** [comment]
```
Thx for the fast response :)

I still got one problem when creating the sqlite db file.
The log file is about 188 Mb big. Extract from the log file:


2015-01-22 21:17:50,408 DEBUG Number revisions converted : 54511 (Rev no : 
54511)
2015-01-22 21:17:50,408 DEBUG Revision author:1srv_anfrank
2015-01-22 21:17:50,408 DEBUG Revision date:2007-12-14 07:30:52
2015-01-22 21:17:50,408 DEBUG Revision msg:Incremented version by automatic 
build at 2007-12-14-08:30:54
2015-01-22 21:17:50,408 DEBUG Changed path count : 1
2015-01-22 21:17:50,408 DEBUG isDirectory: path 
/ProjectViews/Autobuild/SWEPFramework/current/BuildSWEPFRamework.ini revno 54512
2015-01-22 21:17:50,408 DEBUG Trying to get file information for 
file:///D:/Subversion/Development.Chap/ProjectViews/Autobuild/SWEPFramework/curr
ent/BuildSWEPFRamework.ini
2015-01-22 21:17:50,486 DEBUG 54512 : 
/ProjectViews/Autobuild/SWEPFramework/current/BuildSWEPFRamework.ini : M : 0 : 
0 
2015-01-22 21:17:50,486 DEBUG Number revisions converted : 54512 (Rev no : 
54512)
2015-01-22 21:17:50,486 DEBUG Revision author:1srv_anfrank
2015-01-22 21:17:50,486 DEBUG Revision date:2007-12-14 07:30:54
2015-01-22 21:17:50,486 DEBUG Revision msg:Tagging during automatic build at 
2007-12-14-08:30:54
2015-01-22 21:17:50,486 DEBUG Changed path count : 1
2015-01-22 21:17:50,486 DEBUG isDirectory: path 
/ProjectViews/Autobuild/SWEPFramework/tags/V3.5.0.3_2007-12-14-08:30:54 revno 
54513
2015-01-22 21:17:50,486 ERROR Found Error
Traceback (most recent call last):
  File "D:\Subversion_Apps\Subversion_statistics\Python27\Lib\site-packages\svnplot\svnlog2sqlite.py", line 60, in convert
    self.ConvertRevs(startrevno, endrevno, bUpdLineCount)
  File "D:\Subversion_Apps\Subversion_statistics\Python27\Lib\site-packages\svnplot\svnlog2sqlite.py", line 141, in ConvertRevs
    addedfiles, changedfiles, deletedfiles = revlog.changedFileCount()
  File "D:\Subversion_Apps\Subversion_statistics\Python27\Lib\site-packages\svnplot\svnlogiter.py", line 382, in changedFileCount
    isdir = change.isDirectory()
  File "D:\Subversion_Apps\Subversion_statistics\Python27\Lib\site-packages\svnplot\svnlogiter.py", line 119, in isDirectory
    return(self.pathtype() == 'D')
  File "D:\Subversion_Apps\Subversion_statistics\Python27\Lib\site-packages\svnplot\svnlogiter.py", line 189, in pathtype
    self.__updatePathType()
  File "D:\Subversion_Apps\Subversion_statistics\Python27\Lib\site-packages\svnplot\svnlogiter.py", line 90, in __updatePathType
    if(self.logclient.isDirectory(revno, filepath) ==True):
  File "D:\Subversion_Apps\Subversion_statistics\Python27\Lib\site-packages\svnplot\svnlogclient.py", line 507, in isDirectory
    entry = self.getInfo(changepath, revno)
  File "D:\Subversion_Apps\Subversion_statistics\Python27\Lib\site-packages\svnplot\svnlogclient.py", line 400, in getInfo
    url = self.getUrl(path)
  File "D:\Subversion_Apps\Subversion_statistics\Python27\Lib\site-packages\svnplot\svnlogclient.py", line 598, in getUrl
    url = self.getRootUrl() + urllib.pathname2url(path)
  File "D:\Subversion_Apps\Subversion_statistics\Python27\lib\nturl2path.py", line 58, in pathname2url
    raise IOError, error
IOError: Bad path: 
/ProjectViews/Autobuild/SWEPFramework/tags/V3.5.0.3_2007-12-14-08:30:54
2015-01-22 21:17:50,517 INFO Root url found : 
file:///D:/Subversion/Development.Chap
2015-01-22 21:17:50,517 DEBUG Trying to get head revision 
rooturl:file:///D:/Subversion/Development.Chap
2015-01-22 21:17:50,533 DEBUG Found head revision 740570
2015-01-22 21:17:50,533 DEBUG Trying to get revision log. revno:1, 
url=file:///D:/Subversion/Development.Chap
2015-01-22 21:17:50,533 INFO Repository Start-End Rev no : 1-740570
2015-01-22 21:17:50,533 INFO Converting revisions 54511 to 740570
2015-01-22 21:17:50,533 INFO Conversion started
2015-01-22 21:17:50,533 INFO Updating revision from 54511 to 740570
2015-01-22 21:17:50,533 INFO updating logs 54511 to 740570
2015-01-22 21:17:50,533 DEBUG Trying to get revision logs [54511:740570]
2015-01-22 21:17:50,658 DEBUG Revision author:1srv_anfrank
2015-01-22 21:17:50,658 DEBUG Revision date:2007-12-14 07:22:02
2015-01-22 21:17:50,658 DEBUG Revision msg:Created folder remotely
2015-01-22 21:17:50,658 DEBUG Changed path count : 1
2015-01-22 21:17:50,658 DEBUG isDirectory: path 
/ProjectViews/Autobuild/SWEPFramework/tags revno 54511
2015-01-22 21:17:50,658 DEBUG Trying to get file information for 
file:///D:/Subversion/Development.Chap/ProjectViews/Autobuild/SWEPFramework/tags
2015-01-22 21:17:50,752 DEBUG path /ProjectViews/Autobuild/SWEPFramework/tags 
is Directory
2015-01-22 21:17:50,752 DEBUG 54511 : 
/ProjectViews/Autobuild/SWEPFramework/tags/ : A : 0 : 0 
2015-01-22 21:17:50,752 DEBUG Number revisions converted : 1 (Rev no : 54511)
2015-01-22 21:17:50,752 DEBUG Revision author:1srv_anfrank
2015-01-22 21:17:50,752 DEBUG Revision date:2007-12-14 07:30:52
2015-01-22 21:17:50,752 DEBUG Revision msg:Incremented version by automatic 
build at 2007-12-14-08:30:54
2015-01-22 21:17:50,752 DEBUG Changed path count : 1
2015-01-22 21:17:50,752 DEBUG isDirectory: path 
/ProjectViews/Autobuild/SWEPFramework/current/BuildSWEPFRamework.ini revno 54512
2015-01-22 21:17:50,752 DEBUG Trying to get file information for 
file:///D:/Subversion/Development.Chap/ProjectViews/Autobuild/SWEPFramework/curr
ent/BuildSWEPFRamework.ini
2015-01-22 21:17:50,783 DEBUG 54512 : 
/ProjectViews/Autobuild/SWEPFramework/current/BuildSWEPFRamework.ini : M : 0 : 
0 
2015-01-22 21:17:50,783 DEBUG Number revisions converted : 2 (Rev no : 54512)
2015-01-22 21:17:50,783 DEBUG Revision author:1srv_anfrank
2015-01-22 21:17:50,783 DEBUG Revision date:2007-12-14 07:30:54
2015-01-22 21:17:50,783 DEBUG Revision msg:Tagging during automatic build at 
2007-12-14-08:30:54
2015-01-22 21:17:50,783 DEBUG Changed path count : 1
2015-01-22 21:17:50,783 DEBUG isDirectory: path 
/ProjectViews/Autobuild/SWEPFramework/tags/V3.5.0.3_2007-12-14-08:30:54 revno 
54513
2015-01-22 21:17:50,783 ERROR Found Error
Traceback (most recent call last):
  File "D:\Subversion_Apps\Subversion_statistics\Python27\Lib\site-packages\svnplot\svnlog2sqlite.py", line 60, in convert
    self.ConvertRevs(startrevno, endrevno, bUpdLineCount)
  File "D:\Subversion_Apps\Subversion_statistics\Python27\Lib\site-packages\svnplot\svnlog2sqlite.py", line 141, in ConvertRevs
    addedfiles, changedfiles, deletedfiles = revlog.changedFileCount()
  File "D:\Subversion_Apps\Subversion_statistics\Python27\Lib\site-packages\svnplot\svnlogiter.py", line 382, in changedFileCount
    isdir = change.isDirectory()
  File "D:\Subversion_Apps\Subversion_statistics\Python27\Lib\site-packages\svnplot\svnlogiter.py", line 119, in isDirectory
    return(self.pathtype() == 'D')
  File "D:\Subversion_Apps\Subversion_statistics\Python27\Lib\site-packages\svnplot\svnlogiter.py", line 189, in pathtype
    self.__updatePathType()
  File "D:\Subversion_Apps\Subversion_statistics\Python27\Lib\site-packages\svnplot\svnlogiter.py", line 90, in __updatePathType
    if(self.logclient.isDirectory(revno, filepath) ==True):
  File "D:\Subversion_Apps\Subversion_statistics\Python27\Lib\site-packages\svnplot\svnlogclient.py", line 507, in isDirectory
    entry = self.getInfo(changepath, revno)
  File "D:\Subversion_Apps\Subversion_statistics\Python27\Lib\site-packages\svnplot\svnlogclient.py", line 400, in getInfo
    url = self.getUrl(path)
  File "D:\Subversion_Apps\Subversion_statistics\Python27\Lib\site-packages\svnplot\svnlogclient.py", line 598, in getUrl
    url = self.getRootUrl() + urllib.pathname2url(path)
  File "D:\Subversion_Apps\Subversion_statistics\Python27\lib\nturl2path.py", line 58, in pathname2url
    raise IOError, error
IOError: Bad path: 
/ProjectViews/Autobuild/SWEPFramework/tags/V3.5.0.3_2007-12-14-08:30:54
2015-01-22 21:17:50,799 INFO Root url found : 
file:///D:/Subversion/Development.Chap
2015-01-22 21:17:50,799 DEBUG Trying to get head revision 
rooturl:file:///D:/Subversion/Development.Chap
2015-01-22 21:17:50,799 DEBUG Found head revision 740570
2015-01-22 21:17:50,799 DEBUG Trying to get revision log. revno:1, 
url=file:///D:/Subversion/Development.Chap
2015-01-22 21:17:50,799 INFO Repository Start-End Rev no : 1-740570
2015-01-22 21:17:50,799 INFO Converting revisions 54511 to 740570
2015-01-22 21:17:50,799 INFO Conversion started
2015-01-22 21:17:50,799 INFO Updating revision from 54511 to 740570
2015-01-22 21:17:50,799 INFO updating logs 54511 to 740570
2015-01-22 21:17:50,799 DEBUG Trying to get revision logs [54511:740570]
2015-01-22 21:17:50,892 DEBUG Revision author:1srv_anfrank
2015-01-22 21:17:50,892 DEBUG Revision date:2007-12-14 07:22:02
2015-01-22 21:17:50,892 DEBUG Revision msg:Created folder remotely
2015-01-22 21:17:50,892 DEBUG Changed path count : 1
2015-01-22 21:17:50,892 DEBUG isDirectory: path 
/ProjectViews/Autobuild/SWEPFramework/tags revno 54511
2015-01-22 21:17:50,892 DEBUG Trying to get file information for 
file:///D:/Subversion/Development.Chap/ProjectViews/Autobuild/SWEPFramework/tags
2015-01-22 21:17:50,954 DEBUG path /ProjectViews/Autobuild/SWEPFramework/tags 
is Directory
2015-01-22 21:17:50,954 DEBUG 54511 : 
/ProjectViews/Autobuild/SWEPFramework/tags/ : A : 0 : 0 
2015-01-22 21:17:50,954 DEBUG Number revisions converted : 1 (Rev no : 54511)
2015-01-22 21:17:50,954 DEBUG Revision author:1srv_anfrank
2015-01-22 21:17:50,954 DEBUG Revision date:2007-12-14 07:30:52
2015-01-22 21:17:50,954 DEBUG Revision msg:Incremented version by automatic 
build at 2007-12-14-08:30:54
2015-01-22 21:17:50,954 DEBUG Changed path count : 1
2015-01-22 21:17:50,954 DEBUG isDirectory: path 
/ProjectViews/Autobuild/SWEPFramework/current/BuildSWEPFRamework.ini revno 54512
2015-01-22 21:17:50,954 DEBUG Trying to get file information for 
file:///D:/Subversion/Development.Chap/ProjectViews/Autobuild/SWEPFramework/curr
ent/BuildSWEPFRamework.ini
2015-01-22 21:17:51,002 DEBUG 54512 : 
/ProjectViews/Autobuild/SWEPFramework/current/BuildSWEPFRamework.ini : M : 0 : 
0 
2015-01-22 21:17:51,002 DEBUG Number revisions converted : 2 (Rev no : 54512)
2015-01-22 21:17:51,002 DEBUG Revision author:1srv_anfrank
2015-01-22 21:17:51,002 DEBUG Revision date:2007-12-14 07:30:54
2015-01-22 21:17:51,002 DEBUG Revision msg:Tagging during automatic build at 
2007-12-14-08:30:54
2015-01-22 21:17:51,002 DEBUG Changed path count : 1
2015-01-22 21:17:51,002 DEBUG isDirectory: path 
/ProjectViews/Autobuild/SWEPFramework/tags/V3.5.0.3_2007-12-14-08:30:54 revno 
54513
2015-01-22 21:17:51,002 ERROR Found Error
Traceback (most recent call last):
  File "D:\Subversion_Apps\Subversion_statistics\Python27\Lib\site-packages\svnplot\svnlog2sqlite.py", line 60, in convert
    self.ConvertRevs(startrevno, endrevno, bUpdLineCount)
  File "D:\Subversion_Apps\Subversion_statistics\Python27\Lib\site-packages\svnplot\svnlog2sqlite.py", line 141, in ConvertRevs
    addedfiles, changedfiles, deletedfiles = revlog.changedFileCount()
  File "D:\Subversion_Apps\Subversion_statistics\Python27\Lib\site-packages\svnplot\svnlogiter.py", line 382, in changedFileCount
    isdir = change.isDirectory()
  File "D:\Subversion_Apps\Subversion_statistics\Python27\Lib\site-packages\svnplot\svnlogiter.py", line 119, in isDirectory
    return(self.pathtype() == 'D')
  File "D:\Subversion_Apps\Subversion_statistics\Python27\Lib\site-packages\svnplot\svnlogiter.py", line 189, in pathtype
    self.__updatePathType()
  File "D:\Subversion_Apps\Subversion_statistics\Python27\Lib\site-packages\svnplot\svnlogiter.py", line 90, in __updatePathType
    if(self.logclient.isDirectory(revno, filepath) ==True):
  File "D:\Subversion_Apps\Subversion_statistics\Python27\Lib\site-packages\svnplot\svnlogclient.py", line 507, in isDirectory
    entry = self.getInfo(changepath, revno)
  File "D:\Subversion_Apps\Subversion_statistics\Python27\Lib\site-packages\svnplot\svnlogclient.py", line 400, in getInfo
    url = self.getUrl(path)
  File "D:\Subversion_Apps\Subversion_statistics\Python27\Lib\site-packages\svnplot\svnlogclient.py", line 598, in getUrl
    url = self.getRootUrl() + urllib.pathname2url(path)
  File "D:\Subversion_Apps\Subversion_statistics\Python27\lib\nturl2path.py", line 58, in pathname2url
    raise IOError, error
IOError: Bad path: 
/ProjectViews/Autobuild/SWEPFramework/tags/V3.5.0.3_2007-12-14-08:30:54

For me it looks like the path is corrupt!? Can this repo be fixed? Or this 
failure be skipped?

________________________________________________________________________________
_____________________________________________________

The error "UnicodeEncodeError: 'ascii' codec can't encode characters" is fixed.
I imported manually different modules 
"six"       => six-1.9.0-py2.py3-none-any
"dateutil"  => python_dateutil-2.4.0-py2.py3-none-any
"pyparsing" =>pyparsing-2.0.3-py2.py3-none-any.whl

Thx in advance.




```

Original comment by `Har00n...@gmail.com` on 23 Jan 2015 at 8:27

**Comment by GoogleCodeExporter** [comment]
```
>>For me it looks like the path is corrupt!? Can this repo be fixed? Or this 
failure be skipped?

I think the repo is correct. I think its my mistake/bug in the way I am using 
pathname2url function or in the pathname to url function itself. I suspect it 
is because of ':' charactor in the filename.
```

Original comment by `nitinbh...@gmail.com` on 1 Feb 2015 at 2:57

**Comment by GoogleCodeExporter** [comment]
```
Problem is filename/tagname  contains ':' character. It is a valid filename on 
'unix' but not on windows. Since you are running svnplot on windows it is 
'pathname2url' function is giving errors. I have to think of some work around 
for this problem.
```

Original comment by `nitinbh...@gmail.com` on 1 Feb 2015 at 3:28

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision 68273ff68302.
```

Original comment by `nitinbh...@gmail.com` on 1 Feb 2015 at 3:36
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
I have pushed a fix for this in the repository. Not released yet.Check the diff 
for revision 68273ff68302. Its just online change. see if it fixes the issue 
for you.
```

Original comment by `nitinbh...@gmail.com` on 1 Feb 2015 at 3:38
